borsh/src/de/mod.rs

```diff
@@ -42,6 +44,10 @@ pub trait BorshDeserialize: Sized {
     fn deserialize_reader<R: Read>(reader: &mut R) -> Result<Self>;
 
     /// Deserialize this instance from a slice of bytes.
+    #[deprecated(
+        since = "1.0.0",
+        note = "please use `borsh::from_slice` new function instead"
+    )]
```

Comment:
I think we don't need to mark it as deprecated, but make all the examples use `borsh::from_*` methods.

Let's follow [`serde_json`](https://docs.rs/serde_json/latest/serde_json/#functions) as a role model. See that deserialization is not discoverable in [borsh docs](https://docs.rs/borsh/latest/borsh/) these days, and we want to fix that with the helpers. Let's also expose `from_reader` helper.